apps/core/declarativewebcontainer.cpp

```diff
@@ -1040,7 +1042,7 @@ void DeclarativeWebContainer::updateLoading()
 
 void DeclarativeWebContainer::updateActiveTabRendered()
 {
-    if (m_webPage && !m_webPage->completed() && !m_webPage->domContentLoaded()) {
+    if (!m_webPage || !(m_webPage->completed() || m_webPage->domContentLoaded() || m_webPage->isPainted())) {
```

Comment:
In practise completed is always true as it requires that page initialized and it cannot be rendered without being initialized. Still prerequisite for these two other variables.

The latter part of the condition (in the brackets) evaluates always to false as inside brackets is always true. This return statement should be reached when we do not have webpage and neither domContentLoaded nor isPainted is true.